### DISCUSSION TITLE
Does entry.dependOn work with target node?

### DISCUSSION THREAD

**Discussion by rplss** [webpack/webpack#17642] [General]
have a small koa app bundled with webpack@5
the config may like

```
{
    target: "node",
    entry: {
        koa: ["koa", "@koa/router"],
        index: {
            import: path.resolve(__dirname, "../src/index.ts"),
            dependOn: "koa"
        }
    },
    output: {
        ...
        filename: "[name].js"
    },
}
```

the build is ok and `index.js` works fine
the webpack runtime code is in `koa.js`
`index.js` import `koa.js` with require like `var __webpack_require__ = require("./koa.js");`

but if there are more chunks like
```
{
    target: "node",
    entry: {
        koa: "koa",
        router: "@koa/router",
        index: {
            import: path.resolve(__dirname, "../src/index.ts"),
            dependOn: ["koa", "router"]
        }
    }
}
```

the build is also ok but throw error while running `index.js`
this time both `koa.js` and `router.js` have its own runtime
and `__webpack_require__.X(0, ["router"],...)` will not find chunk `router.js` in `koa.js`
`GetChunkFilenameRuntimeModule` returns `undefined` in both runtime

the runtime find neither `koa.js` nor `router.js` while set `optimization.runtimeChunk` to `single` 

it works fine if use `optimization.splitChunks` instead of `entry.dependOn`
`GetChunkFilenameRuntimeModule` will return `${chunkId}.js` and runtime could get the chunk
but the dependences of `koa` and `@koa/router` are bundled in `index.js` which is a little unacceptable

any ideas

**Comment by alexander-akait** [comment]
Sorry for delay, is it still valid? If yes, can you create reproducible test repo?
